### ISSUE TITLE
Syntax highlighting gets confused by optionality of whitespace

### ISSUE THREAD

**Issue by ssokolow** [DNSGeek/vim-cbmbasic#1]
I just decided to start playing around with the [Commodore 64 Programmer's Reference Guide](https://www.commodore.ca/commodore-manuals/commodore-64-programmers-reference-guide/) that I got in a lot of tech hand-me-downs years ago, and I noticed that your syntax definitions get confused by the optionality of whitespace in CBM BASIC, as demonstrated by this example I chose to type in at random (from page 200):

![scan](https://user-images.githubusercontent.com/46915/105665340-bc475f00-5ece-11eb-9a92-f01bebccc8ad.png)
![Spectacle nS8802](https://user-images.githubusercontent.com/46915/105665481-16e0bb00-5ecf-11eb-9279-89cd0e6f4dfe.png)

**Comment by DNSGeek** [comment]
Yeah, it does. I see no easy way to distinguish (e.g.) "FOR I = " and "FORI=". The latter looks like a variable declaration. I don't think I can fix this, you'll just have to use whitespace to get highlighted words. Sorry.
